akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/AccountExampleOneLinersInModel.java

```diff
@@ -0,0 +1,164 @@
+            return new OpenedAccount(0.0);
+        }
+
+    }
+
+    public class OpenedAccount implements Account {
+        public final double balance;
+
+        OpenedAccount(double balance) {
+            this.balance = balance;
+        }
+
+        Effect<AccountEvent, Account> depositCommand(Deposit deposit) {
+            return Effect().persist(new Deposited(deposit.amount));
+        }
+
+        Effect<AccountEvent, Account> withdrawCommand(Withdraw withdraw) {
+            if ((balance - withdraw.amount) < 0.0) {
+                return Effect().unhandled(); // TODO replies are missing in this example
+            } else {
+                return Effect().persist(new Withdrawn(withdraw.amount))
+                        .andThen(acc2 -> { // FIXME in scaladsl it's named thenRun, change javadsl also?
+                            // we know this cast is safe, but somewhat ugly
+                            OpenedAccount openAccount = (OpenedAccount) acc2;
+                            // do some side-effect using balance
+                            System.out.println(openAccount.balance);
+                        });
+            }
+        }
+
+        Effect<AccountEvent, Account> closeCommand(CloseAccount cmd) {
+            if (balance == 0.0)
+                return Effect().persist(new AccountClosed());
+            else
+                return Effect().unhandled();
+        }
+
+        OpenedAccount makeDeposit(Deposited deposit) {
+            return new OpenedAccount(balance + deposit.amount);
+        }
+
+        OpenedAccount makeWithdraw(Withdrawn withdrawn) {
+            return new OpenedAccount(balance - withdrawn.amount);
+        }
+
+        ClosedAccount closeAccount(AccountClosed cmd) {
+            return new ClosedAccount();
+        }
+    }
+
+    public class ClosedAccount implements Account {
+    }
+
+    public static Behavior<AccountCommand> behavior(String accountNumber) {
+        return Behaviors.setup(context -> new AccountExampleOneLinersInModel(context, accountNumber));
+    }
+
+    public AccountExampleOneLinersInModel(ActorContext<AccountCommand> context, String accountNumber) {
+        super(accountNumber);
+    }
+
+    @Override
+    public Account emptyState() {
+        return new EmptyAccount();
+    }
+
+
+    @Override
+    public void commandHandlerSetup(HandlerBuilders.CommandHandlerBuilder<AccountCommand, AccountEvent, Account> builder) {
+
+        builder.forType(EmptyAccount.class)
+                .matchCommand(CreateAccount.class, EmptyAccount::createAccount);
+
+        builder.forType(OpenedAccount.class)
+                .matchCommand(Deposit.class, OpenedAccount::depositCommand)
+                .matchCommand(Withdraw.class, OpenedAccount::withdrawCommand)
+                .matchCommand(CloseAccount.class, OpenedAccount::closeCommand);
+
+        builder.forType(ClosedAccount.class)
+                .matchCommand(AccountCommand.class, () -> Effect().unhandled());
+
+    }
```

Comment:
Here is it in action. 

The user is forced to implement this method and only needs to declare the cases. 

Return is not needed, since mutable.

Reply:
This looks very good!

Reply:
Good, I will continue and add the same for event handlers.

Reply:
I think this shows the benefits of a mutable builder, which supports adding several `forStateType` cases with readable alignments and new-line separators between the cases. Great!